packages/core/src/telemetry/service.ts

```diff
@@ -78,6 +161,27 @@ export class TelemetryService {
     return hash.digest("hex");
   }
 
+  private flushDetached() {
+    const serializedEvents = JSON.stringify(this.events);
+    const fileName = `${this.distDir}/telemetry-events.json`;
+    fs.writeFileSync(fileName, serializedEvents);
+  }
+
+  private async getContext() {
+    if (this.context) {
+      return this.context;
+    }
+
+    this.context = {
+      sessionId: this.sessionId,
+      projectId: await this.getProjectId(),
+      anonymousId: this.anonymousId,
+      meta: this.anonymousMeta,
+    };
```

Comment:
Don't think we need private getters for anonymousId and anonymousMeta, because they will only ever be used here. Can inline them. This means we can also get rid of the `metadata` class property.

If it were me, I would also inline getProjectId, and get rid of the `rawProjectId` class property.

Reply:
that's a great call actually - I do think we can keep `anonymousId` because it uses the persisted value from `conf` similar to `salt` and provide a readable way to say "hey, get this from the file otherwise set it"